Core/Core/Transports/SQLite.cs

```diff
@@ -257,8 +257,10 @@ public void UpdateObject(string hash, string serializedObject)
   {
     CancellationToken.ThrowIfCancellationRequested();
 
+    using var c = new SqliteConnection(_connectionString);
```

Comment:
Our AccountManager does some potentially sketchy operations calling `UpdateObject` while looping through the lazily enumerate return of `GetAllObjects`. This causes `GetAllObjects` to return forever, since it returns your updated objects also.

Reverting my change so `GetAllObjects` and `UpdateObject` use separate SQLite connections appears to fix things.
I'm a little concerned that we're actually relying on some sort of in-memory cache that the connections use.

I question if this is the behaviour we actually want, it seems to me the problem is with the way the `AccountManager` is enumerating over a lazy enumerable that its updating. I'll open a ticket for next release to decide on what our desired behaviour here should be.

Reply:
I agree. We should do some investigation as to why this is the way it works now, as this sounds super dodgy...